connectors/kafka/src/main/java/forklift/controller/AcknowledgedRecordHandler.java

```diff
@@ -0,0 +1,121 @@
+package forklift.controller;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Maintains a batch of acknowledged records and provides management for adding and removing {@link org.apache.kafka.common.TopicPartition partitions}.
+ * Acknowledged records are those that have started processing but have not yet been committed to the Kafka Broker.  This class is threadsafe.
+ */
+public class AcknowledgedRecordHandler {
+    private static final Logger log = LoggerFactory.getLogger(AcknowledgedRecordHandler.class);
+    private ConcurrentHashMap<TopicPartition, OffsetAndMetadata> pendingOffsets = new ConcurrentHashMap<>();
+    private Set<TopicPartition> assignment = ConcurrentHashMap.newKeySet();
+    private ReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
I find your use of `ReadWriteLock` a little confusing -- I guess you're just using it to protect the non-atomic operations of `removePartitions`?

If that's the case, a little documentation of that among all of this documentation would be nice.